### ISSUE TITLE
How do I build with MS Build?

### ISSUE THREAD

**Issue by Kurren123** [fsharp/fsharp#782]
I have a .net framework class library project with a single `Game.fs` file containing this:

```
module Game

type Mario = Mario of string
module Mario =
    let sayHi = "Its'a me, Maaario!"
```

This builds using visual studio. I try `MSBuild.exe /p:VisualStudioVersion=12.0` (the visual studio version as per [this answer](https://stackoverflow.com/a/20095260/1282944)) however I am getting back the following error:

```
Duplicate definition of type, exception or module 'Mario'
``` 

I am guessing that when I use MS Build it is not passing in the `ModuleSuffix` [compiler flag](https://msdn.microsoft.com/en-us/visualfsharpdocs/conceptual/core.compilationrepresentationflags-enumeration-%5Bfsharp%5D). How can I pass this in, and preferably replicate the build that visual studio does?

Thanks for any help given!

**Comment by dsyme** [comment]
I think there are differences in the F# compiler you are using in each case - the second (which is giving the error) is an older verison

**Comment by dsyme** [comment]
@Kurren123 Could you outline the exact versions of Visual Studio and /or Mono and/or OS you are using?

**Comment by Kurren123** [comment]
Hi @dsyme sorry for the delay. Versions:
```
Visual Studio 2017 Enterprise 
Version 15.2 (26430.16) Release
Visual F# 4.1   00369-60000-00001-AA955
Microsoft Visual F# 4.1
Windows 10
```
I would use fsc.exe but I see that visual studio passes tons of flags so there doesn't seem a straightforward way.

**Comment by dsyme** [comment]
Passing ``/p:VisualStudioVersion=12.0`` uses an old version of the F# compiler.  Why are you passing that?  You don't need to I think.

Perhaps you need to adjust your project file

**Comment by Kurren123** [comment]
Hi @dsyme  thanks for your answer. I uploaded my minimal test project [here](https://github.com/Kurren123/FsharpMsBuildTest). I removed `/p:VisualStudioVersion=12.0` but its still giving back the same error. 

After looking at visual studio's output, it uses the CoreCompile `C:\Program Files (x86)\Microsoft SDKs\F#\4.1\Framework\v4.0\fsc.exe`

Whereas MSBuild 14 at path `C:\Program Files (x86)\MSBuild\14.0\Bin\amd64` is using CoreCompile `C:\Program Files (x86)\Microsoft SDKs\F#\4.0\Framework\v4.0\fsc.exe`. 

Could this be a mismatch of the MS build versions that visual studio uses? Or is there a setting somewhere to set the fsc version? Thanks again for your help

**Comment by Kurren123** [comment]
Okay after looking around I found out that my project was using MS Build 15 due to this in my fsproj file: `<Project ToolsVersion="15.0"`. I thought the highest version on my machine was 14 but it turns out that MSBuild 15 is in a different location from the rest of them: `C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\Bin\amd64`. It builds successfully with MSBuild 15!

Closing this as resolved
